WooCommerce/Classes/ViewRelated/Products/Edit Product/ProductFormViewController.swift

```diff
@@ -200,6 +198,27 @@ extension ProductFormViewController: UITableViewDelegate {
     }
 }
 
+// MARK: Action - Edit Product Images
+//
+private extension ProductFormViewController {
+    func showProductImages() {
+        let imagesViewController = ProductImagesViewController(product: product) { [weak self] images in
+            self?.onEditProductImagesCompletion(images: images)
+        }
+        navigationController?.pushViewController(imagesViewController, animated: true)
+    }
+
+    func onEditProductImagesCompletion(images: [ProductImage]) {
+        defer {
```

Comment:
What does defer do? At first I thought it was used here to prevent the need for a closure in the `imagesUpdated` method, but all that's doing is returning a new struct as the update.

Reply:
`defer` is to perform the block of code whichever way the code exits. In this case, the code could exit either from the`guard` else block or the update block, and `defer` code is called either way at the end. here's an [NSHipster post on guard/defer](https://nshipster.com/guard-and-defer/)